NavigationReactNative/src/ios/NVNavigationBarView.m

```diff
@@ -12,6 +13,14 @@ - (id)init
     return self;
 }
 
+- (void)removeReactSubview:(UIView *)subview {
+    [super removeReactSubview:subview];
+    
+    if ([subview isKindOfClass:NVBarTitleView.class]) {
+        self.reactViewController.navigationItem.titleView = nil;
+    }
+}
```

Comment:
Let’s clear the `titleView` property using the `willMoveToSuperview` method of the `NVBarTitleView`. Then the `BarTitleViewIOS` component doesn’t have to be the child of the `NavigationBarIOS` component. I did something similar in the [NVSearchBarView](https://github.com/grahammendick/navigation/blob/753ecc6ad8c3735d583ccd2baa76a12239a5e19e/NavigationReactNative/src/ios/NVSearchBarView.m#L104)

Reply:
👍 This makes code more clear. There's a little layout issue after this change. I'll push the change after figure it out.

Reply:
In the current approach,`BarTitleViewIOS` style affects `NavigationBarIOS` layout properties. In some cases this leads to taking some space in content view.

For example, after adding `BarTitleViewIOS` to [`sample/zoom/Detail.js`](https://github.com/grahammendick/navigation/blob/master/NavigationReactNative/sample/zoom/Detail.js):

### Screenshot:
![layout-issue-screenshot](https://user-images.githubusercontent.com/13307997/64318898-b9a04d00-cfed-11e9-8922-43c4b09db730.png)

### Hierarchy layers:
![layout-issue-hierarchy](https://user-images.githubusercontent.com/13307997/64318905-be650100-cfed-11e9-8b55-fc0499184511.png)

Still struggling resolving this. Please comment here if you have any idea.

Reply:
Does the problem go away if you set a `width`?

Reply:
No 😞, the `BarTitleViewIOS` is a child of `NavigationBarIOS`. React Native does the layout calculation work according to the virtual dom hierarchy relation. So `NavigationBarIOS`'s `width` & `height` is affected.

Reply:
I tried adding the `BarTitleViewIOS` to the Details of zoom and it works well. The Hierarchy layers in your picture look correct to me. Are you saying it's wrong because you can't see the back button text?

Reply:
The hierarchy is correct, but the `NVNavigationBarView` is expected to be 0 height so that it's invisible. 
In my experiment, the height of `NVNavigationBarView` is affected by the `TitleView`.

Reply:
Oh, thanks. I see what you mean! Sorry, I didn't look at your pictures clearly enough. Well spotted!

Reply:
We can fix this by setting a height of 0 on the `NavigationBarIOS`.
```jsx
<NavigationBarIOS title="Color" style={{height: 0}}>
```
We shouldn't make users do this though. We should set this style inside the [NavigationBarIOS component](https://github.com/grahammendick/navigation/blob/master/NavigationReactNative/src/NavigationBarIOS.tsx) so that it's hidden from the user.

Reply:
We can also fix it by setting position to 'absolute` instead of height to 0
```jsx
<NavigationBarIOS title="Color" style={{position: 'absolute'}}> 
```
Which one do you prefer?
Whichever we choose, we have to hide this away inside the `NavigationBarIOS component` so that the user isn't impacted

Reply:
Cool ! I prefer `position: 'absolute'` one, let me try it.

Reply:
Just realised, we can set `position: absolute` on `BarTitleViewIOS` instead! Even better because we don't have to change `NavigationBarIOS`, right?

Reply:
Yeah, it's a better choice.